### PR TITLE
FISH-1204 Missing example values in OpenApi components schema definition using nested objects

### PR DESCRIPTION
**Pull request by aubi**
## Description
Fix missing description of properties of embedded objects, fix missing required flag for reference, issue https://github.com/payara/Payara/issues/5151

### New tests
Added one new test for dependent DTOs with @Schema annotation, one for required reference.

### Testing Performed
Tested on /openapi -- contains full description of embedded properties and a required reference is referred.
The correct output is attached to the GH issue.

### CODE REVIEW

**Comment by aubi** [comment]
jenkins tests please

**Comment by aubi** [comment]
TCK test succeeded

File: appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/model/media/SchemaImpl.java [hunk 1]
```diff
@@ -133,20 +133,26 @@ public static SchemaImpl valueOf(String content) throws JsonMappingException, Js
                 .readValue(content, SchemaImpl.class);
     }
 
-    @SuppressWarnings("unchecked")
+    //@SuppressWarnings("unchecked")
```
Comment (lprimak) [inline appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/model/media/SchemaImpl.java:136]: coding standards - remove dead code not comment it out

File: appserver/payara-appserver-modules/microprofile/openapi/src/test/java/fish/payara/microprofile/openapi/test/app/application/DependantClassesTest.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
+/*
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+ *
+ * Copyright (c) [2019] Payara Foundation and/or its affiliates. All rights reserved.
```
Comment (lprimak) [inline appserver/payara-appserver-modules/microprofile/openapi/src/test/java/fish/payara/microprofile/openapi/test/app/application/DependantClassesTest.java:4]: copyright 2021

File: appserver/payara-appserver-modules/microprofile/openapi/src/test/java/fish/payara/microprofile/openapi/test/app/application/RequiredAttrTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
+/*
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+ *
+ * Copyright (c) [2019] Payara Foundation and/or its affiliates. All rights reserved.
```
Comment (lprimak) [inline appserver/payara-appserver-modules/microprofile/openapi/src/test/java/fish/payara/microprofile/openapi/test/app/application/RequiredAttrTest.java:4]: 2021

**Review by lprimak — CHANGES_REQUESTED** [review]
just small nit-picks, otherwise LGTM

**Review by lprimak — APPROVED** [review]

**Comment by lprimak** [comment]
jenkins test

**Pull request merged**